test/test_gemv_device.cc

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) 2017-2023, University of Tennessee. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#include "test.hh"
+#include "cblas_wrappers.hh"
+#include "lapack_wrappers.hh"
+#include "blas/flops.hh"
+#include "print_matrix.hh"
+#include "check_gemm.hh"
+
+// -----------------------------------------------------------------------------
+template <typename TA, typename TX, typename TY>
+void test_gemv_device_work( Params& params, bool run )
+{
+    using namespace testsweeper;
+    using std::abs, std::real, std::imag;
+    using blas::Op, blas::Layout, blas::max;
+    using scalar_t = blas::scalar_type< TA, TX, TY >;
+    using real_t   = blas::real_type< scalar_t >;
+
+    // get & mark input values
+    blas::Layout layout = params.layout();
+    blas::Op trans  = params.trans();
+    scalar_t alpha  = params.alpha.get<scalar_t>();
+    scalar_t beta   = params.beta.get<scalar_t>();
+    int64_t m       = params.dim.m();
+    int64_t n       = params.dim.n();
+    int64_t incx    = params.incx();
+    int64_t incy    = params.incy();
+    int64_t device  = params.device();
+    int64_t align   = params.align();
+    int64_t verbose = params.verbose();
+
+    // mark non-standard output values
+    params.gflops();
+    params.gbytes();
+    params.ref_time();
+    params.ref_gflops();
+    params.ref_gbytes();
+
+    // adjust header to msec
+    params.time.name( "time (ms)" );
+    params.ref_time.name( "ref time (ms)" );
+    params.ref_time.width( 13 );
```

Comment:
I think `width` is no longer needed here. TestSweeper automatically adjusts upward to the width of `name`. Let's fix in a later PR.